datasource/src/main/scala/quasar/physical/mongo/SSL.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2014–2019 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.physical.mongo
+
+import slamdata.Predef._
+
+import cats.effect.Sync
+import cats.syntax.functor._
+import cats.syntax.flatMap._
+
+import java.io.ByteArrayInputStream
+import java.security.{SecureRandom, KeyStore}
+import java.security.cert.{X509Certificate, CertificateFactory}
+import javax.net.ssl.{SSLContext, KeyManager, TrustManager, X509TrustManager, KeyManagerFactory, TrustManagerFactory}
+
+object SSL {
+  def context[F[_]: Sync](config: SSLConfig): F[SSLContext] = for {
+    ctx <- Sync[F].delay(SSLContext.getInstance("TLS"))
+    ks <- keyManagers[F](config)
+    ts <- trustManagers[F](config)
+    rnd <- Sync[F].delay(new SecureRandom())
+    _ <- Sync[F].delay(ctx.init(ks, ts, rnd))
+  } yield ctx
+
+  @SuppressWarnings(Array("org.wartremover.warts.Null"))
+  def keyManagers[F[_]: Sync](config: SSLConfig): F[Array[KeyManager]] = config.clientPEM match {
```

Comment:
I don't mind usually but since this can return `null` because of java lib integration, I prefer functions like this to be private.

Reply:
:+1: